ml-agents/mlagents/trainers/torch/utils.py

```diff
@@ -29,6 +29,66 @@ def swish(input_activation: torch.Tensor) -> torch.Tensor:
         """Swish activation function. For more info: https://arxiv.org/abs/1710.05941"""
         return torch.mul(input_activation, torch.sigmoid(input_activation))
 
+    @staticmethod
+    def apply_learning_rate(optim: torch.optim.Optimizer, lr: float) -> None:
+        """
+        Apply a learning rate to a torch optimizer.
+        :param optim: Optimizer
+        :param lr: Learning rate
+        """
+        for param_group in optim.param_groups:
+            param_group["lr"] = lr
+
+    @staticmethod
+    def get_decayed_parameter(
```

Comment:
Would it make more sense to instantiate a DecayingParamter(torch.Parameter) that keeps the schedule, max_step, min_value, initial_value, etc... and that has an additional "update" method that takes the global_step (since it is the only thing that changes.

Reply:
I like this idea, since it would naturally save via state_dict. Currently it would have to go through the `update` at least once before the right `lr` is set.

Reply:
Hmm, on further thinking, there doesn't seem to be a real reason to make it an actual Parameter object though, since it won't be backpropp'ed through nor saved to an `nn.Module` (it's outside in the Optimizer). We can make it a DecayValue object with a `.get_value(global_step)`, and that should cut down the large function calls.

Reply:
`optim.lr_scheduler`?

Reply:
+1

Reply:
`optim.lr_scheduler`?

Reply:
I didn't go with `optim.lr_scheduler` since a) the API assumes lr is a function of the number of epochs, and we want to do it based on the number of steps, and b) `beta` and `epsilon` would still need another solution. Not really sure what the right answer is here. Just an `Object` seems strange but that seems to be all that is needed.